src/backend/distributed/commands/index.c

```diff
@@ -585,6 +695,90 @@ RangeVarCallbackForDropIndex(const RangeVar *rel, Oid relOid, Oid oldRelOid, voi
 }
 
 
+/*
+ * Check permissions on table before acquiring relation lock; also lock
+ * the heap before the RangeVarGetRelidExtended takes the index lock, to avoid
+ * deadlocks.
+ *
+ * This code is borrowed from RangeVarCallbackForReindexIndex() in
+ * commands/indexcmds.c in Postgres source. We need this to ensure the right
+ * order of locking while dealing with REINDEX statements.
+ */
+static void
+RangeVarCallbackForReindexIndex(const RangeVar *relation, Oid relId, Oid oldRelId,
+								void *arg)
+{
+	/* *INDENT-OFF* */
+	char		relkind;
+	struct ReindexIndexCallbackState *state = arg;
+	LOCKMODE	table_lockmode;
+
+	/*
+	 * Lock level here should match table lock in reindex_index() for
+	 * non-concurrent case and table locks used by index_concurrently_*() for
+	 * concurrent case.
+	 */
+#if PG_VERSION_NUM >= 120000
+	table_lockmode = state->concurrent ? ShareUpdateExclusiveLock : ShareLock;
+#else
+	table_lockmode = ShareLock;
+#endif
+
+	/*
+	 * If we previously locked some other index's heap, and the name we're
+	 * looking up no longer refers to that relation, release the now-useless
+	 * lock.
+	 */
+	if (relId != oldRelId && OidIsValid(oldRelId))
+	{
+		UnlockRelationOid(state->locked_table_oid, table_lockmode);
+		state->locked_table_oid = InvalidOid;
+	}
+
+	/* If the relation does not exist, there's nothing more to do. */
+	if (!OidIsValid(relId))
```

Comment:
Just glanced over this code and noticed there are some if statements without enclosing braces `{` `}`.

https://github.com/citusdata/wiki/wiki/Development-Team-&-Process#coding-conventions

I realise now that the coding conventions are still in a google doc which not everyone might have access to.

relevant section:

```
We add curly braces after each if / else. Even if the block is only one statement:

if (queryResultList != NIL)
{
	succeededPlacementList = lappend(succeededPlacementList, 
                                       shardPlacement);
}
else
{
	failedPlacementList = lappend(failedPlacementList,
                                    shardPlacement);
}
```

Lets make sure we move this document to OneDrive.

No priority now, but I think there are a couple of action items for the team here:
 - uncrustify should catch this, there is no benefit in humans checking coding style which computers can do
 - go back through the code to make sure everything conforms, which is easily done if uncrustify has a rule for this.
 - once everyone has access to the styleguide promote reading it :)

Reply:
Note that the code in question has INDENT-OFF because it's copied from postgres

Reply:
We should add a note on copying postgres code to the style guide. We do so ad verbatim with INDENT-OFF to be able to diff later.

Reply:
ah, makes sense